R/make-constructor.R

```diff
@@ -0,0 +1,217 @@
+#' @include layer.R
+#' @include scale-type.R
+NULL
+
+#' Produce boilerplate constructors
+#'
+#' The `make_constructor()` functions sets up a user-facing constructor for
+#' ggproto classes. Currently, `make_constructor()` is implemented for
+#' `Geom` classes.
+#'
+#' @param x An object to setup a constructor for.
+#' @param ... Name-value pairs to use as additional arguments in the
+#'   constructor. For layers, these are passed on to [`layer(params)`][layer()].
+#' @param checks A list of calls to be evaluated before construction of the
+#'   object, such as one constructed with [`exprs()`][rlang::exprs()].
```

Comment:
Can you add an example of this to the @examples as I don't think it is immediately obvious to most users how to use it